### PR TITLE
Allow to track queue change in follow-ups

### PR DESCRIPTION
**Pull request by samsplunks**
Because of Time calculation feature that allows for time in specific queues not to be counted in, we need to track queue changes in TicketChanges.

As ticket changes are related to follow-ups, I have added the ability to change ticket queue in the "Change Further Details" of the ticket view.

If this is approved, it will enable me to code correct calculation for the FOLLOWUP_TIME_SPENT_EXCLUDE_QUEUES setting.

In my use case, queues are used to pass tickets to different teams.

### CODE REVIEW

**Review by uhurusurfa — COMMENTED** [review]
This change seems reasonable to me.
Could you add a small unit test for this additional feature (maybe in helpdesk/tests/test_ticket_actions.py)?

**Comment by samsplunks** [comment]
Thanks, yes I will and I will also update the time calculation code relative to each follow-up.

The idea is that if a helpdesk admin wants to introduce time_spent auto calculation or wish to modify the calculation settings, he can make a queryset with all FollowUps, remove time_spent and .save() to (re)calculate the time.

**Comment by samsplunks** [comment]
I have refactored the time_spent_calculation so it doesn't make unnecessary queries when no exclusion is set.

It is now dealing with queue changes over time with the newly introduced TicketChange for field _('Queue').

A unit test was made to validate creation of the TicketChange together with the modification of the ticket queue.

Another one to test multiple queue exclusions over time with correct time_spent calculation.

**Review by uhurusurfa — APPROVED** [review]
Looks good - thanks for adding the unit tests

**Comment by uhurusurfa** [comment]
@Benbb96 - if you do not see any issues here then please merge the PR

File: helpdesk/update_ticket.py [hunk 1]
```diff
@@ -302,6 +305,16 @@ def update_ticket(
...
         ticket.priority = priority
 
+    if queue != ticket.queue.id:
+        c = TicketChange(
+            followup=f,
+            field=_('Queue'),
+            old_value=ticket.queue.id,
+            new_value=queue,
+        )
+        c.save()
```
Comment (Benbb96) [inline helpdesk/update_ticket.py:315]: Do you think you could take the opportunity here to refactor all the creation of TicketChange like this :

```suggestion
        c = f.ticketchange_set.create(
            field=_('Queue'),
            old_value=ticket.queue.id,
            new_value=queue,
        )
```

**Comment by samsplunks** [comment]
Thank you @Benbb96, suggestions committed!

**Review by Benbb96 — APPROVED** [review]
Alright, no problem!

**Pull request merged**